app/models/course_user.rb

```diff
@@ -33,7 +33,7 @@ def experience_points
       sum(:points_awarded)
     end
   end
-  has_one :invitation, class_name: Course::UserInvitation.name, dependent: :destroy
+  has_one :invitation, class_name: Course::UserInvitation.name, dependent: :destroy, validate: true
```

Comment:
@lowjoel I find that there is a bi-directional `dependent: :destroy` on `course_user` and `user_invitation`,  both of them cannot be destroyed in this case. Can you fix this as well ?